sh_scrapy/commands/shub_image_info.py

```diff
@@ -35,8 +36,21 @@ def add_options(self, parser):
     def run(self, args, opts):
         result = {
             'project_type': 'scrapy',
-            'spiders': sorted(self.crawler_process.spider_loader.list())
+            'spiders': sorted(self.crawler_process.spider_loader.list()),
+            'metadata': {},
         }
+        if sys.version_info >= (3, 8):
+            # scrapy-spider-metadata requires Python 3.8+
+            from scrapy_spider_metadata import get_metadata_for_spider
+            for spider_name in result['spiders']:
+                spider_cls = self.crawler_process.spider_loader.load(spider_name)
+                metadata_dict = get_metadata_for_spider(spider_cls)
+                try:
+                    # make sure it's serializable
+                    json.dumps(metadata_dict)
+                except (TypeError, ValueError):
+                    continue
```

Comment:
should we have a warning here?

Reply:
The command prints a JSON text which some other components read, we could print warnings to stderr but I don't think anything will read them (but I'm not sure).

Reply:
Still, it seems currently it's the only place where the error could be show; without it there is no indication of issue at all. But maybe there is a better place for it?

Reply:
I tried to do that and found that the command explicitly disables logging: https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/blob/71aacd898a6b49604bf89c582e878f83c48a173a/sh_scrapy/commands/shub_image_info.py#L11

Reply:
We might need to find another place to log these errorrs, or document it (in scrapy-spider-metadata?). But it seems this shouldn't block this PR.